src/check_release.rs

```diff
@@ -150,6 +150,20 @@ fn print_triggered_lint(
                 Ok(())
             })?;
         }
+
+        if let Some(hint_template) = semver_query.witness.hint_template.as_deref() {
+            let message = config
+                .handlebars()
+                .render_template(hint_template, &pretty_result)
+                .context("Error instantiating witness hint template.")?;
+
+            config.log_info(|config| {
+                config.shell_note("the following downstream code would break:")?;
+                writeln!(config.stderr(), "{message}")?;
```

Comment:
I'm a bit worried about readability here. It's too early to fuss over spacing and the exact CLI output format, but just wanted to flag this for us to revisit as a polish item before shipping this feature.

Reply:
In terms of spacing between items, or the messages themselves (or both?)  I did get syntax highlighting (on the other branch) working for code snippets so that could help for the snippets themselves.

Reply:
Spacing, mostly, both between the note and the code, and possible indentation of the hint code. We want to do our best to be readable but concise, and match cargo/rustc style to make merging into cargo easier.

It'll be easier to revisit this later once we have some more witnesses that we're showing. For now, let's leave them hidden behind some permanently-unstable CLI flag (e.g. `-Zunstable-options --show-witness-hints`) so we can iterate freely?

Perhaps adding the infrastructure for such unstable flags would be a good PR in and of itself?